### ISSUE TITLE
There is a compatibility problem of DubboHeadersGetter#keys in Dubbo 2.7.6 and 2.7.7

### ISSUE THREAD

**Issue by steverao** [open-telemetry/opentelemetry-java-instrumentation#12978]
### Describe the bug

In Dubbo instrumentation, I found a compatibility problem of [DubboHeadersGetter#keys](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d69fd1a4a7dcc1530e01b2d51541ae8784324e1f/instrumentation/apache-dubbo-2.7/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/apachedubbo/v2_7/DubboHeadersGetter.java#L16) in Dubbo 2.7.6, 2.7.7. 
In related versions, the `getAttachments` is as follows:
```java
@Deprecated
@Override
public Map<String, String> getAttachments() {
    return new AttachmentsAdapter.ObjectToStringMap(attachments);
}

public static class ObjectToStringMap extends HashMap<String, String> {
        private Map<String, Object> attachments;

        public ObjectToStringMap(Map<String, Object> attachments) {
            this.attachments = attachments;
        }

        @Override
        public String get(Object key) {
            Object obj = attachments.get(key);
            return convert(obj);
        }

        @Override
        public String put(String key, String value) {
            Object obj = attachments.put(key, value);
            return convert(obj);
        }

        @Override
        public String remove(Object key) {
            Object obj = attachments.remove(key);
            return convert(obj);
        }

        private String convert(Object obj) {
            if (obj instanceof String) {
                return (String) obj;
            }
            return null; // or JSON.toString(obj);
        }
    }
```
The `ObjectToStringMap` can't get all keySet by `keySet` method, so it can't get correct result of DubboHeadersGetter#keys in related version.

### Steps to reproduce

We can compare the codes between Dubbo 2.7.6, 2.7.7 and others, we can get above conclusion by codes.

### Expected behavior

DubboHeadersGetter#keys can't get correct result in Dubbo 2.7.6, 2.7.7

### Actual behavior

DubboHeadersGetter#keys can get correct result in Dubbo 2.7.6, 2.7.7

### Javaagent or library instrumentation version

2.11.0

**Comment by laurit** [comment]
> The ObjectToStringMap doesn't contain keySet method, so it can't get correct result of DubboHeadersGetter#keys in related version.

I don't get this. `ObjectToStringMap` extends `HashMap` that has `keySet` method. Elements added by `super.put` that is called in the constructor of `ObjectToStringMap` should be in that set. The implementation for `ObjectToStringMap` is different for versions 2.7.6 and 2.7.7 where `super.put` is not called in the constructor for `ObjectToStringMap`.

**Comment by steverao** [comment]
> I don't get this. `ObjectToStringMap` extends `HashMap` that has `keySet` method. Elements added by `super.put` that is called in the constructor of `ObjectToStringMap` should be in that set. The implementation for `ObjectToStringMap` is different for versions 2.7.6 and 2.7.7 where `super.put` is not called in the constructor for `ObjectToStringMap`.

Yes, as your said, there are something wrong in previous description, I have corrected them.

**Comment by laurit** [comment]
> Yes, as your said, there are something wrong in previous description, I have corrected them.

I think the title of the issue could also be changed to indicate that the issue is present only in a specific versions. Currently it leaves an impression that there is a problem in all versions later and including 2.7.6.

**Comment by steverao** [comment]
> I think the title of the issue could also be changed to indicate that the issue is present only in a specific versions. Currently it leaves an impression that there is a problem in all versions later and including 2.7.6.

Sure, done
